graphqlapi/src/test/java/io/stargate/graphql/schema/GraphQlTestBase.java

```diff
@@ -81,27 +69,66 @@ public void setupEnvironment() {
     // a bit overkill and brittle. The integration tests in the 'testing' module fill that gap.
     ResultSet resultSet = mock(ResultSet.class);
     when(resultSet.rows()).thenReturn(Collections.emptyList());
-    when(dataStore.query(queryCaptor.capture()))
-        .thenReturn(CompletableFuture.completedFuture(resultSet));
+
+    try {
+      String roleName = "mock role name";
+      when(authenticationService.validateToken(token)).thenReturn(storedCredentials);
+      when(storedCredentials.getRoleName()).thenReturn(roleName);
+      dataStoreCreateMock = mockStatic(DataStore.class);
+      dataStoreCreateMock
+          .when(() -> DataStore.create(eq(persistence), eq(roleName), parametersCaptor.capture()))
+          .then(
+              i -> {
+                DataStore dataStore = mock(DataStore.class);
+                when(dataStore.query(queryCaptor.capture()))
+                    .thenReturn(CompletableFuture.completedFuture(resultSet));
+
+                // Batches use multiple data store instances, one per each mutation
+                // We need to capture the parameters provided at dataStore creation
+                Parameters dataStoreParameters = i.getArgument(2, Parameters.class);
+                when(dataStore.batch(batchCaptor.capture()))
+                    .then(
+                        batchInvoke -> {
+                          batchParameters = dataStoreParameters;
+                          return CompletableFuture.completedFuture(mock(ResultSet.class));
+                        });
+                return dataStore;
+              });
+    } catch (Exception e) {
+      fail("Unexpected exception while mocking dataStore", e);
+    }
 
     graphQlSchema = createGraphQlSchema();
-    graphQl = GraphQL.newGraphQL(graphQlSchema).build();
+    graphQl =
+        GraphQL.newGraphQL(graphQlSchema)
+            // Use parallel execution strategy for mutations (serial is default)
+            .mutationExecutionStrategy(new AsyncExecutionStrategy())
```

Comment:
should we test both? `AsyncExecutionStrategy` and serial? Maybe a parameterized test?

Reply:
No, we are forcing `AsyncExecutionStrategy` for all DML mutations (see `CustomGraphQLServlet`), so only that strategy will be used for DML mutations.